src/main.rs

```diff
@@ -99,10 +99,7 @@ fn main() {
 
     let cmdline = cmd_arguments.value_of("cmdline");
 
-    let disks: Vec<&str> = cmd_arguments
-        .values_of("disk")
-        .expect("Missing argument: disk. Provide at least one")
-        .collect();
+    let disks: Option<Vec<&str>> = cmd_arguments.values_of("disk").map(|x| x.collect());
```

Comment:
We need to verify that we either have a pmem backed file or a disk. Maybe have a `validate()` routine on `VmParams` to check that our final configuration is valid.

Reply:
When using direct kernel boot you could have a situation where you don't want any local disk to e.g.  using an embedded initramfs or just to test the kernel. However I feel that it's more likely that people would accidentally forget to include either --disk or --pmem and so validation is a good idea.